File: src/main/java/io/alpere/common/crudfop/repository/BaseRepository.java
package io.alpere.common.crudfop.repository;


import io.alpere.common.crudfop.model.BaseEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.querydsl.QuerydslPredicateExecutor;

import java.util.UUID;

public interface BaseRepository<Entity extends BaseEntity>
        extends JpaRepository<Entity, UUID>, QuerydslPredicateExecutor<Entity> {
}

File: src/main/kotlin/io/alpere/crud/repository/BaseRepository.kt
package io.alpere.crud.repository

import io.alpere.crud.model.BaseEntity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.querydsl.QuerydslPredicateExecutor
import java.util.*

interface BaseRepository<Entity: BaseEntity>: JpaRepository<Entity, UUID>, QuerydslPredicateExecutor<Entity>